src/colorizer/TimeControls.ts

```diff
@@ -81,22 +86,34 @@ export default class TimeControls {
     this.timerId = window.setTimeout(() => loadNextFrame(this.canvas.getCurrentFrame()), 0);
   }
 
-  public async handlePlayButtonClick(): Promise<void> {
+  /**
+   * Begins playback of the time series. If the time series is already playing, this function does
+   * nothing.
+   * @param onNewFrameCallback An optional callback that will be called whenever a new frame is loaded.
+   */
+  public async play(onNewFrameCallback: () => void = () => {}): Promise<void> {
     if (this.canvas.getCurrentFrame() >= this.canvas.getTotalFrames() - 1) {
       await this.canvas.setFrame(0);
     }
-    this.playTimeSeries(() => {});
+    this.playTimeSeries(onNewFrameCallback);
   }
 
-  public handlePauseButtonClick(): void {
-    if (this.timerId !== NO_TIMER_ID) {
+  /**
+   * Pauses the playback of the time series. If any pause listeners have been added, their callbacks
+   * will be triggered.
+   */
+  public pause(): void {
+    if (this.isPlaying()) {
       clearTimeout(this.timerId);
     }
     this.timerId = NO_TIMER_ID;
     this.pauseCallbacks.forEach((callback) => callback());
   }
 
-  public async handleFrameAdvance(delta: number = 1): Promise<void> {
+  /**
+   * Increment or decrement the time series by the given number of frames.
+   */
+  public async offsetFrame(delta: number = 1): Promise<void> {
```

Comment:
Nit: I think advanceFrame or incrementFrame might be slightly better.  The word offset for me can have other array, indexing, or memory implications.